core/api/serializers/project.py

```diff
@@ -52,8 +66,76 @@ class Meta:
             "plus",
         ]
 
+    def validate_category(self, value):
+        categories = [
+            x[0] for x in ProjectSubmission.ProjectSubmissionCategories.choices
+        ]
+        if value not in categories:
+            raise serializers.ValidationError("Invalid category")
+        return value
```

Comment:
If you use a [ChoiceField](https://www.django-rest-framework.org/api-guide/fields/#choicefield) instead of a `CharField` you should not have need to perform this validation manually. 

However, I think (but am not sure) that DRF will automatically use a `ChoiceField` if you simply don't specify a custom serializer field for `category` at all. It should also mark it as required as long as it's doesn't have `null=True` in the model.